eslint-rules/tests/lib/rules/typography-deprecation.js

```diff
@@ -439,6 +519,194 @@ ruleTester.run('typography-deprecation', rule, {
+        import React, {Component} from 'react';
+        import {Typography, Text} from 'our-source';
+        export default class OurList extends Component {
+          render() {
+            const {isTitle} = this.props;
+            const titleVal = this.props.isTitle;
+            const subtitleVal = !this.props.isTitle;
+            return (
+              <Text title={titleVal} subtitle={subtitleVal}/>
+            )
+          }
+        }`,
+        errors: [{message: `'Typography.title' is deprecated. Please use 'Typography.heading' instead (fix is available).`}]
+    },
+    {
+      options: options,
+      code: `
+        import React, {Component} from 'react';
+        import {Typography, Text} from 'our-source';
+        export default class OurList extends Component {
+          render() {
+            const {isTitle} = this.props;
+            const titleVal = isTitle;
+            const subtitleVal = !isTitle;
+            return (
+              <Text title={titleVal} subtitle={subtitleVal}/>
+            )
+          }
+        }`,
+        errors: [{message: `'Typography.title' is deprecated. Please use 'Typography.heading' instead (fix is available).`}]
+    },
+    {
+      options: options,
+      code: `
+        import React, {Component} from 'react';
+        import {Typography, TextField} from 'our-source';
+        export default class OurList extends Component {
+          render() {
+            const titleVal = true;
+            return (
+              <TextField title={titleVal}/>
+            )
+          }
+        }`,
+        errors: [{message: `'Typography.title' is deprecated. Please use 'Typography.heading' instead (fix is available).`}]
+    },
+    {
+      options: options,
+      code: `
+        import React, {Component} from 'react';
+        import {Typography, TextField} from 'our-source';
+        export default class OurList extends Component {
+          render() {
+            const {isTitle} = this.props;
+            const titleVal = this.props.isTitle;
+            const subtitleVal = !this.props.isTitle;
+            return (
+              <TextField title={titleVal} subtitle={subtitleVal}/>
+            )
+          }
+        }`,
+        errors: [{message: `'Typography.title' is deprecated. Please use 'Typography.heading' instead (fix is available).`}]
+    },
+    {
+      options: options,
+      code: `
+        import React, {Component} from 'react';
+        import {Typography, TextField} from 'our-source';
+        export default class OurList extends Component {
+          render() {
+            const {isTitle} = this.props;
+            const titleVal = isTitle;
+            const subtitleVal = !isTitle;
+            return (
+              <TextField title={titleVal} subtitle={subtitleVal}/>
+            )
+          }
+        }`,
+        errors: [{message: `'Typography.title' is deprecated. Please use 'Typography.heading' instead (fix is available).`}]
+    },
+    // {
```

Comment:
Why are we keeping these? 
These are the use cases we don't support, right? 
Maybe write a comment that explain these tests and that we can't/don't support them

Reply:
Added comments (TODOs)